endtoend/evaluators/operations.go

```diff
@@ -0,0 +1,231 @@
+	Policy:     afterNthEpoch(22),
+	Evaluation: depositedValidatorsAreActive,
+}
+
+// ProposeVoluntaryExit sends a voluntary exit from randomly selected validator in the genesis set.
+var ProposeVoluntaryExit = types.Evaluator{
+	Name:       "propose_voluntary_exit_epoch_%d",
+	Policy:     onEpoch(5),
+	Evaluation: proposeVoluntaryExit,
+}
+
+// ValidatorHasExited checks the beacon state for the exited validator and ensures its marked as exited.
+var ValidatorHasExited = types.Evaluator{
+	Name:       "voluntary_has_exited_%d",
+	Policy:     onEpoch(6),
+	Evaluation: validatorIsExited,
+}
+
+// Not including first epoch because of issues with genesis.
+func isBetweenEpochs(fromEpoch uint64, toEpoch uint64) func(uint64) bool {
+	return func(currentEpoch uint64) bool {
+		return fromEpoch < currentEpoch && currentEpoch > toEpoch
+	}
+}
+
+func processesDepositedValidators(conns ...*grpc.ClientConn) error {
+	conn := conns[0]
+	client := eth.NewBeaconChainClient(conn)
+	validatorRequest := &eth.ListValidatorsRequest{
+		PageSize:  int32(params.BeaconConfig().MinGenesisActiveValidatorCount),
+		PageToken: "1",
+	}
+	validators, err := client.ListValidators(context.Background(), validatorRequest)
+	if err != nil {
+		return errors.Wrap(err, "failed to get validators")
+	}
+
+	expectedCount := params.BeaconConfig().MinGenesisActiveValidatorCount / uint64(e2e.TestParams.BeaconNodeCount)
+	receivedCount := uint64(len(validators.ValidatorList))
+	if expectedCount != receivedCount {
+		return fmt.Errorf("expected validator count to be %d, recevied %d", expectedCount, receivedCount)
+	}
+
+	churnLimit, err := helpers.ValidatorChurnLimit(params.BeaconConfig().MinGenesisActiveValidatorCount + uint64(len(validators.ValidatorList)))
+	if err != nil {
+		return errors.Wrap(err, "failed to calculate churn limit")
+	}
+	effBalanceLowCount := 0
+	activeEpoch10Count := 0
+	activeEpoch11Count := 0
+	activeEpoch12Count := 0
+	activeEpoch13Count := 0
+	exitEpochWrongCount := 0
+	withdrawEpochWrongCount := 0
+	for _, item := range validators.ValidatorList {
+		if item.Validator.EffectiveBalance < params.BeaconConfig().MaxEffectiveBalance {
+			effBalanceLowCount++
+		}
+		if item.Validator.ActivationEpoch == 10 {
+			activeEpoch10Count++
+		} else if item.Validator.ActivationEpoch == 11 {
+			activeEpoch11Count++
+		} else if item.Validator.ActivationEpoch == 12 {
+			activeEpoch12Count++
+		} else if item.Validator.ActivationEpoch == 13 {
+			activeEpoch13Count++
+		}
+
+		if item.Validator.ExitEpoch != params.BeaconConfig().FarFutureEpoch {
+			exitEpochWrongCount++
+		}
+		if item.Validator.WithdrawableEpoch != params.BeaconConfig().FarFutureEpoch {
+			withdrawEpochWrongCount++
+		}
+	}
+
+	if effBalanceLowCount > 0 {
+		return fmt.Errorf(
+			"%d validators did not have genesis validator effective balance of %d",
+			effBalanceLowCount,
+			params.BeaconConfig().MaxEffectiveBalance,
+		)
+	} else if activeEpoch10Count != int(churnLimit) {
```

Comment:
same here, switch statement?

Reply:
Can't really make it work for this one but worked great for the first one, thanks :+1: